mcs/class/corlib/System/MulticastDelegate.cs

```diff
@@ -61,13 +60,20 @@ public override void GetObjectData (SerializationInfo info, StreamingContext con
 			base.GetObjectData  (info, context);
 		}
 
-
 		protected sealed override object DynamicInvokeImpl (object[] args)
 		{
-			if (prev != null)
-				prev.DynamicInvokeImpl (args);
-
-			return base.DynamicInvokeImpl (args);
+			if (delegates == null) {
+				return base.DynamicInvokeImpl (args);
+			} else {
+				int len = delegates.Length;
+				if (len >= 1) {
+					object r = null;
+					for (int i = 0; i < len; ++i)
+						r = delegates [i].DynamicInvoke (args);
+					return r;
+				}
+				throw new InvalidOperationException ("delegates should not be empty");
```

Comment:
the "3" states a multicast delegate can be are:
 - no delegate: the reference is null (not really a state)
 - 1 delegate : the data is stored in the Delegate fields (method_ptr, invoke_impl, ....), but the field delegates is null
 - 2+ delegates : the fields in Delegate are null, but delegates is non null

In the case of 1 delegate, the delegates field is null because we do not call the contructor on MulticastDelegate.